packages/labs/analyzer/src/lib/javascript/variables.ts

```diff
@@ -134,10 +136,15 @@ const getVariableDeclarationInfoList = (
       )
       .flat();
   } else {
-    throw new DiagnosticsError(
-      dec,
-      `Expected declaration name to either be an Identifier or a BindingPattern`
+    analyzer.diagnostics.push(
+      createDiagnostic({
+        node: dec,
+        message: `Expected declaration name to either be an Identifier or a BindingPattern`,
```

Comment:
What's a normal-person for a binding pattern? Is that an destructuring assignment statement?